gnoblog-cli/cmd/main.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/peterbourgon/ff/v3/ffcli"
+	"os"
+)
+
+type baseCliCfg struct {
+	debug     bool
+	publish   bool
+	edit      bool
+	gasWanted int64
+	gasFee    string
+	chainId   string
+	blogPath  string
+
+	keyName               string
+	gnoHome               string
+	remote                string
+	quiet                 bool
+	insecurePasswordStdIn bool
+}
+
+func main() {
+	var (
+		baseCfg = &baseCliCfg{}
+		fs      = flag.NewFlagSet("root", flag.ExitOnError)
+	)
+
+	// Register the flags
+	baseCfg.registerFlags(fs)
+
+	// Create the root command
+	cmd := &ffcli.Command{
+		ShortUsage: "<subcommand> [flags] [<arg>...] <KEY>",
+		LongHelp:   "The CLI for easy use of the r/blog realm",
+		FlagSet:    fs,
+		Exec: func(_ context.Context, args []string) error {
+			if len(args) != 1 {
+				return errors.New("deployment key not specified")
+			}
+			baseCfg.keyName = args[0]
+			return flag.ErrHelp
+		},
+	}
+
+	// Add the subcommands
+	cmd.Subcommands = []*ffcli.Command{
+		newPostCommand(baseCfg),
+		newBatchCommand(baseCfg),
```

Comment:
I would rather have a batch method that can handle a single one, allowing for a single command. Could you please implement the batch functionality in the "post" command? This way, we can keep the verb "post," which I believe is better, while still supporting multiple posts.

Reply:
Sure. How about I add a `--batch-root` flag that will indicate the posting should be done in batch? The root dir passed in would be searched recursively to find posts inside.